modules/lang-painless/src/main/java/org/opensearch/painless/api/Json.java

```diff
@@ -84,8 +83,6 @@ public static String dump(Object data, boolean pretty) throws IOException {
         }
         builder.value(data);
         builder.flush();
-        try (OutputStream out = builder.getOutputStream()) {
-            return out.toString();
-        }
+        return builder.getOutputStream().toString();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource returned by the following method call: *getOutputStream*. Execution paths that do not contain closure statements were detected. Ensure the object returned by *getOutputStream()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).